### DISCUSSION TITLE
Juicefs objbench fails on gluster backend

### DISCUSSION THREAD

**Discussion by karlkashofer** [juicedata/juicefs#4916] [General]
I tried to do objbench on our gluster backend, and it seems to have a problem:

`karl@sx253:/home/devarea$ sudo juicefs objbench --storage gluster sy081,sy082,sy083/gv0
Start Functional Testing ...
2024/06/02 22:32:26.937953 juicefs[3333534] <ERROR>: list : file descriptor in bad state [object_storage.go:192]
+----------+---------------------+-------------------------------------------+
| CATEGORY |         TEST        |                   RESULT                  |
+----------+---------------------+-------------------------------------------+
|    basic |     create a bucket |                                      pass |
|    basic |       put an object |                                      pass |
|    basic |       get an object |                                      pass |
|    basic |       get non-exist |                                      pass |
|    basic |  get partial object |                                      pass |
|    basic |      head an object |                                      pass |
|    basic |    delete an object |                                      pass |
|    basic |    delete non-exist |                                      pass |
|    basic |        list objects | list failed: file descriptor in bad state |
|     sync |         special key |                                      pass |
|     sync |    put a big object |                                      pass |
|     sync | put an empty object |                                      pass |
|     sync |    multipart upload |                               not support |
|     sync |  change owner/group |      failed to chown object not supported |
|     sync |   change permission |                                      pass |
|     sync |        change mtime |           failed to chtimes not supported |
+----------+---------------------+-------------------------------------------+

Start Performance Testing ...
2024/06/02 22:33:02.498859 juicefs[3333534] <ERROR>: list : invalid argument [object_storage.go:192]
2024/06/02 22:33:02.498898 juicefs[3333534] <ERROR>: list : invalid argument [object_storage.go:192]
2024/06/02 22:33:02.498933 juicefs[3333534] <ERROR>: list : invalid argument [object_storage.go:192]
2024/06/02 22:33:02.498998 juicefs[3333534] <ERROR>: list : invalid argument [object_storage.go:192]
put small objects: 100/100 [================================]        27.7/s          used: 3.612108092s 
get small objects: 100/100 [================================]        450.4/s         used: 222.129251ms 
   upload objects: 256/256 [================================]        15.6/s          used: 16.363811266s
 download objects: 256/256 [================================]        26.3/s          used: 9.731371497s 
     list objects: 0/100 [----------------------------------]  0.0/s         ETA: 0s                    
^C
karl@sx253:/home/devarea$ `
Objbench just hangs at the end, had to force-quit. The test files do appear on the gluster storage and remain there after force-quitting.

The gluster datastore works fine and it can be used for juicefs, i have a filesystem running on it, cant see anything wrong.

Any idea what could be the reason for this error ?

**Comment by zhijian-pro** [comment]
`| basic | list objects | list failed: file descriptor in bad state |`
This error was unexpected, need to test it again.

**Reply by karlkashofer** [reply]
Hi !
Let me know if there is anything i can do to help !
Cheers,
KK
